test/integration/setup/node-attestation/client.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"context"
+	"crypto/rand"
+	x509 "crypto/x509"
+	"fmt"
+	"log"
+	mathrand "math/rand"
+	"time"
+
+	agent "github.com/spiffe/spire/proto/spire/api/server/agent/v1"
+	types "github.com/spiffe/spire/proto/spire/types"
+	"github.com/spiffe/spire/test/integration/setup/itclient"
+	"github.com/spiffe/spire/test/testkey"
+)
+
+var (
+	key = testkey.MustEC256()
+)
+
+func main() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	c := itclient.NewLocalServerClient(ctx)
+	defer c.Release()
+
+	// Validate call to New Downstream X509 CA
+	err := testAgentAPI(ctx, c)
+	if err != nil {
+		log.Fatal(fmt.Sprintf("%v", err))
+	}
+}
+
+func testAgentAPI(ctx context.Context, c *itclient.LocalServerClient) error {
+	csr, err := x509.CreateCertificateRequest(rand.Reader, &x509.CertificateRequest{}, key)
+	if err != nil {
+		return fmt.Errorf("fails to create CSR: %v", err)
+	}
+
+	s1 := mathrand.NewSource(time.Now().UnixNano())
+	tokenID := mathrand.New(s1).Intn(1000000)
+
+	tokenName := fmt.Sprintf("test_token_%v", tokenID)
+
+	// Create a join token using the local socket connection (simulating the CLI running on the spire-server)
+	agentClient := c.AgentClient()
+	_, err = agentClient.CreateJoinToken(ctx, &agent.CreateJoinTokenRequest{Ttl: 1000, Token: tokenName})
```

Comment:
must not you use the token from create join token? (take a look to admin rpcs ITs)

Reply:
This does actually work since the join token is fixed